src/VisualStudio/CSharp/Impl/Options/AutomationObject.cs

```diff
@@ -36,10 +36,25 @@ public int BringUpOnIdentifier
             set { SetBooleanOption(CompletionOptions.TriggerOnTypingLetters, value); }
         }
 
+        // This SettingStore option has now been deprecated in favor of CSharpClosedFileDiagnostics.
         public int ClosedFileDiagnostics
         {
-            get { return GetBooleanOption(ServiceFeatureOnOffOptions.ClosedFileDiagnostic); }
-            set { SetBooleanOption(ServiceFeatureOnOffOptions.ClosedFileDiagnostic, value); }
+            get { return GetIntegerOption(ServiceFeatureOnOffOptions.ClosedFileDiagnostic); }
+            set
+            {
+                // Even though this option has been deprecated, we want to respect the setting if the user has explicitly turned off closed file diagnostics (which is the non-default value for 'ClosedFileDiagnostics').
+                // So, we invoke the setter only for value = 0.
+                if (value == 0)
+                {
+                    SetIntegerOption(ServiceFeatureOnOffOptions.ClosedFileDiagnostic, value);
```

Comment:
It will get set only for profiles which disable ClosedFileDiagnostics - we have lot of profiles which specify true for this option (see [here](http://torch/?q=ClosedFileDiagnostics%20f%3Avssettings&index=Dev14C_20160420))